cron/createFDQNBlockList.py

```diff
@@ -0,0 +1,248 @@
+#                 iplist.append(ipcidrnet)
+#         else:
+#             logger.debug('invalid:' + ip)
+#     return iplist
+
+def parse_fqdn_whitelist(fqdn_whitelist_location):
+    fqdns = []
+    with open(fqdn_whitelist_location, "r") as text_file:
+        for line in text_file:
+            line=line.strip().strip("'").strip('"')
+            if isFQDN(line):
+                fqdns.append(line)
+    return fqdns
+
+def main():
+    logger.debug('starting')
+    logger.debug(options)
+    try:
+        # connect to mongo
+        client = MongoClient(options.mongohost, options.mongoport)
+        mozdefdb = client.meteor
+        fqdnblocklist = mozdefdb['fqdnblocklist']
+        attackers=mozdefdb['attackers']
+        # ensure indexes
+        fqdnblocklist.create_index([('dateExpiring',-1)])
+
+        # TODO: determine if this is relevant in this context
+        # attackers.create_index([('lastseentimestamp',-1)])
+        # attackers.create_index([('category',1)])
+
+        # First, gather IP addresses from recent attackers and add to the block list
+        # attackerIPList = aggregateAttackerIPs(attackers)
+
+        # add attacker IPs to the blocklist
+        # first delete ones we've created from an attacker
+        # ipblocklist.delete_many({'creator': 'mozdef','reference':'attacker'})
+
+        # delete any that expired
+        fqdnblocklist.delete_many({'dateExpiring': {"$lte": datetime.utcnow()-timedelta(days=options.expireage)}})
+
+        # add the aggregations we've found recently
+        # for ip in attackerIPList:
+        #     ipblocklist.insert_one(
+        #         {'_id': genMeteorID(),
+        #          'address':ip,
+        #          'reference': 'attacker',
+        #          'creator':'mozdef',
+        #          'dateAdded': datetime.utcnow()})
+
+        # Lastly, export the combined blocklist
+        fqdnCursor=mozdefdb['fqdnblocklist'].aggregate([
+                {"$sort": {"dateAdded": -1}},
+                {"$match": {"address": {"$exists": True}}},
+                {"$match":
+                    {"$or":[
+                        {"dateExpiring": {"$gte": datetime.utcnow()}},
+                        {"dateExpiring": {"$exists": False}},
+                    ]},
+                },
+                {"$project":{"address":1}},
+                {"$limit": options.iplimit}
+            ])
+        FQDNList=[]
+        for fqdn in fqdnCursor:
+            # TODO: figure out what to do here
+            FQDNList.append(ip['address'])
+        # to text
+        with open(options.outputfile, 'w') as outputfile:
+            for fqdn in FQDNList:
+                outputfile.write("{0}\n".format(fqdn))
+        outputfile.close()
+        # to s3?
+        if len(options.aws_bucket_name)>0:
+            s3_upload_file(options.outputfile, options.aws_bucket_name, options.aws_document_key_name)
+
+
+    except ValueError as e:
+        logger.error("Exception %r generating IP block list" % e)
```

Comment:
s/IP/DNS

Reply:
fixed, went with FQDN for consistency, but the entry point for the user will have FQDN/DNS in the UI, so it's not like a "what is FQDN?" type response.